frontend/src/firebase/firebaseInteractor.ts

```diff
@@ -527,53 +525,46 @@ export default class FirebaseInteractor {
     let incorrect = 0;
     let wordResults: WordResult[] = [];
     let assessmentResultObjects:
-      | firebase.firestore.QuerySnapshot<firebase.firestore.DocumentData>
+      | FirebaseAssessmentResults
       | undefined = undefined;
 
     if (assessmentForSession !== undefined) {
       assessmentDuration =
         Math.round(assessmentForSession.data().durationsInSeconds * 100) / 100;
-      let assessmentResults = await assessmentForSession.ref
-        .collection("results")
-        .get();
+      let assessmentResults = assessmentForSession.data()
+        .results as FirebaseAssessmentResults;
```

Comment:
what happens without the cast here /  in other words? does ts complain ;-;

Reply:
.data() returns `any` since we don't have any knowledge of what data types are in the database from within the code. You can think of this more as an annotation than a cast, so that I can use type hints to see what fields are present on assessmentResults